src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java

```diff
@@ -623,16 +625,22 @@ void updateDirectoryEntry(Firestore firestore, String collectionId, FireStoreDir
     private DocumentSnapshot lookupByPathNoXn(
         Firestore firestore, String collectionId, String lookupPath) throws InterruptedException {
 
-        try {
-            DocumentReference docRef =
-                firestore
-                    .collection(collectionId)
-                    .document(encodePathAsFirestoreDocumentName(lookupPath));
-            ApiFuture<DocumentSnapshot> docSnapFuture = docRef.get();
-            return docSnapFuture.get();
-        } catch (AbortedException | ExecutionException ex) {
-            throw fireStoreUtils.handleExecutionException(ex, "lookupByPathNoXn");
+        RuntimeException lastException = null;
+        for (int retryNum = 0; retryNum < LOOKUP_RETRIES; retryNum++) {
+            try {
+                DocumentReference docRef =
+                    firestore
+                        .collection(collectionId)
```

Comment:
Academic question: I wonder if calling collection actually makes a request to firestore or if it just preps the request...if it's the prior, we may want to move that out of the retry loop

Reply:
Good point! I moved it out of the loop and re-ran my manual test which showed this work on retry.

Reply:
Ok cool!